src/features/island/buildings/lib/craftingMachine.ts

```diff
@@ -144,7 +145,7 @@ export const craftingMachine = createMachine<
 
           const now = Date.now();
 
-          return Math.floor((readyAt - now) / 1000);
+          return (readyAt - now) / 1000;
```

Comment:
This can end up being a floating point number which is why we were flooring it here. Is there a reason you have removed this?

Reply:
The removal of Math.floor() is intentional:
  - It aligns with other components (plots, resource nodes remianing time) where the time left is haldled with a floating point.  The secondsToString() method is going to round the value down anyways in the UI
  - It solve the issue where the progress bar progress is displaying the wrong progress for several amounts of seconds
  - It solve the issue where the "Is it ready yet?" (or something that will replace this modal) modal isn't displaying for the last second in the progress